src/plat/spike/machine/hardware.c

```diff
@@ -140,24 +140,35 @@ ackInterrupt(irq_t irq)
 }
 
 static inline uint64_t get_cycles(void)
-#if __riscv_xlen == 32
 {
+    UNUSED unsigned long mtvec;
+
+    if (config_set(CONFIG_SEL4_RV_MACHINE)) {
+        /* Trap to riscv-pk to get time */
+        mtvec = read_csr_env(tvec);
```

Comment:
nitpick: I would discourage calling local variables by the same name as a machine register as it makes things like `write_csr(mtvec, mtvec)` confusing as the pretense is that 'mtvec' is some kind of magical global symbol and `write_csr` is a function that will write to it. Obviously it is a macro and this code will work, but it's very jarring to read and breaks the illusion that these macros are trying to create. Also it meant I had to dance around language in writing these comments as saying "and later you write mtvec into mtvec" suddenly becomes a bit ambiguous